### PR TITLE
Align the tables we want to use from asm

### PR DESCRIPTION
**Pull request by lu-zero**
- [x] export subpel_filters
- [ ] refactor the concept in a macro for further usage

Should help on #832

### CODE REVIEW

File: src/partition.rs [hunk 1]
```diff
@@ -775,9 +775,13 @@ pub enum MvSubpelPrecision {
   MV_SUBPEL_HIGH_PRECISION
 }
 
+mod subpel_filters {
 pub const SUBPEL_FILTER_SIZE: usize = 8;
 
-const SUBPEL_FILTERS: [[[i32; SUBPEL_FILTER_SIZE]; 16]; 6] = [
+#[repr(C, align(256))]
```
Comment (tdaede) [inline src/partition.rs]: Looks fine, but do you really need 256 byte alignment?

**Comment by lu-zero** [comment]
32 would be enough for all our current usage, I think?

**Comment by KyleSiefring** [comment]
There seems to already be an aligned type in util.rs. https://github.com/xiph/rav1e/blob/master/src/util.rs#L133

**Comment by lu-zero** [comment]
@KyleSiefring yes, we can rework the code to use it in the macro, but I'd rather have you confirm that this approach works while I'm flying back home :)

**Comment by KyleSiefring** [comment]
Does not work. SUBPEL_FILTERS : Aligned = Aligned([ does work. I would like it if we could find some documentation that confirms that this behavior is defined.

**Comment by lu-zero** [comment]
Do you have the code somewhere?

On Jan 2, 2019 16:20, "Kyle Siefring" <notifications@github.com> wrote:

Does not work. SUBPEL_FILTERS : Aligned = Aligned([ does work. I would like
it if we could find some documentation that confirms that this behavior is
defined.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/xiph/rav1e/pull/834#issuecomment-450890855>, or mute
the thread

**Comment by KyleSiefring** [comment]
https://github.com/KyleSiefring/rav1e/tree/asm_data_in_rust_WIP_1

**Comment by KyleSiefring** [comment]
FYI The variable is rav1e_mc_subpel_filters and not SUBPEL_FILTERS in my code.

**Comment by lu-zero** [comment]
https://github.com/rust-av/rav1e/pull/new/asm_data_in_rust_WIP_1

this builds

**Comment by tdaede** [comment]
Do we still need this?

**Pull request closed without merging**